### ISSUE TITLE
AIP-38 | Show historic metrics in Dashboard

### ISSUE THREAD

**Issue by bbovenzi** [apache/airflow#42700]
Once https://github.com/apache/airflow/pull/42629 is merged, we can initialize our new homepage dashboard.

Right now `historical_metrics` only includes total counts. So we can build this part of the Dashboard mockups except for the bar graphs:

![Image](https://github.com/user-attachments/assets/9a99857f-9d43-4d6e-84c3-24d8bc797ca5)

That includes:
- a date range selector
- horizontal bar graphs showing the total count of dag run and task instance states and their overall percentage

We'll ignore dag run type for now.

**Comment by tirkarthi** [comment]
@bbovenzi I can take this up if you are okay. I have prototyped below versions as per design without the checkbox and the bar charts which don't have data from API. I can raise this as PR for the base part and fine tune the UI/UX during review or subsequent PRs.

![image](https://github.com/user-attachments/assets/2e814d6f-fe7a-49ef-9322-cc361090906e)

![image](https://github.com/user-attachments/assets/4414e9bf-e481-4e9d-a0d8-1e325c5e7e73)

**Comment by bbovenzi** [comment]
Nice! One nit, let's keep the phrase "Task Instances". 
We'll probably want to update the historical_metrics endpoint to better fit this view. But that's a separate issue

**Comment by tirkarthi** [comment]
@bbovenzi I am building the asset events section in the dashboard. As part of the dashboard designs the asset events was also to the right of historical metrics so my question is do they belong to the same section and are affected by the time range filter selected? If I select last 1 hour does historical metrics along with the asset events display the last 1 hour metrics only?

https://github.com/apache/airflow/issues/42371#issuecomment-2390830661

**Comment by bbovenzi** [comment]
Yeah, I guess they would be. I'll be honest though, I think the dashboard view is subject to a lot of change from what the designs are now and what we will have when 3.0 comes out.
